### PR TITLE
Add support for Sigstore verification materials

### PR DESCRIPTION
**Pull request by di**
This PR adds support for including the files that comprise Sigstore verification materials with CPython downloads. This is one of the last steps in the [plan documented here](https://docs.google.com/document/d/1_y0O7mbv4TVdw3rx9tUCrIZItA9GGOA-RNH_qQW1JS8/edit?resourcekey=0-vsbpfJONHGpYf3AfytbD8g).

~This also fixes some small issues I ran into during the development process.~ moved to #2114 

(cc @ewdurbin @pablogsal @Yhg1s)

### CODE REVIEW

File: templates/humans.txt
Comment (ewdurbin) [inline templates/humans.txt]: nit: we're also on heroku now and use gunicorn 🙃

File: sponsors/migrations/0086_auto_20220809_1655.py
Comment (ewdurbin) [inline sponsors/migrations/0086_auto_20220809_1655.py]: exclude this file for now please, I'll add a followup PR to include this dangling migration.

File: templates/downloads/release_detail.html
Comment (ewdurbin) [inline templates/downloads/release_detail.html]: Would it be a bad idea to link from this to some reference on what sigstore is?

File: templates/downloads/release_detail.html
Comment (ewdurbin) [inline templates/downloads/release_detail.html]: I don't think this is valid for Django templates:

```
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 43: 'set', expected 'endblock'. Did you forget to register or load this tag?
```

File: base-requirements.txt
Comment (ewdurbin) [inline base-requirements.txt]: is this strictly necessary? could it be addressed in a separate hygiene PR?

File: .python-version
Comment (ewdurbin) [inline .python-version]: is this strictly necessary? could it be addressed in a separate hygiene PR?

File: templates/downloads/release_detail.html
Comment (di) [reply templates/downloads/release_detail.html]: That's the plan! I'll get that page up before we merge this, and add the link here.

File: base-requirements.txt
Comment (ewdurbin) [reply base-requirements.txt]: Just a note on why: I'm not fully certain how this affects deployment pipeline and don't want it to impact the ability to safely merge this.

File: .python-version
Comment (di) [reply .python-version]: Moved this to https://github.com/python/pythondotorg/pull/2114

File: base-requirements.txt
Comment (di) [reply base-requirements.txt]: Moves this to https://github.com/python/pythondotorg/pull/2114

File: sponsors/migrations/0086_auto_20220809_1655.py
Comment (di) [reply sponsors/migrations/0086_auto_20220809_1655.py]: Moved this to https://github.com/python/pythondotorg/pull/2114.

**Review by ewdurbin — APPROVED** [review]

**Comment by ewdurbin** [comment]
cc @python/python-release-managers, take a look if you're able before I merge this tomorrow.

**Comment by pablogsal** [comment]
Will this need changes to our upload script (https://github.com/python/release-tools/blob/master/add-to-pydotorg.py) ?

**Comment by di** [comment]
@pablogsal Looks like yes, I was unaware of that script, will make a PR to update it tomorrow.

**Comment by di** [comment]
(shouldn't block this PR though I think)

**Comment by di** [comment]
@pablogsal PR here: https://github.com/python/release-tools/pull/17

**Comment by ewdurbin** [comment]
Was it determined that this won't break the existing release script? If so I'm good to merge!

**Comment by di** [comment]
Yes, the changes to the release script are to include the verification materials if they're present. As is, it just won't include them, but won't break.

**Comment by pablogsal** [comment]
@di Seems that we need some extra changes because after signing 3.10.7 with sigstore we ran into some problems:

https://www.python.org/downloads/release/python-3107/

If you click the sigstore cert it shows 403 Forbidden.

Also, unfortunately there was a NameError in the original changes for the release scripts:

https://github.com/python/release-tools/pull/19

Also, seems that the automation is broken now because sigstore blocks until you paste some code from the auth and therefore this code here:

https://github.com/python/release-tools/blob/401afdec47eadf4b5f17a68f5bb87f4ca04a86d1/run_release.py#L720-L744

Doesn't work anymore.  Sadly I will need to comment out the signing in the `add_to_python_org.py` script until that is fixed.

**Comment by ewdurbin** [comment]
Are the 403ing files on disk on the downloads backend host? Seems they must be being created with strict permissions?

**Comment by pablogsal** [comment]
Yeah, they are on disk. Seems that the changes in the release tools so far are not enough.  The release script changes the permissions for the artifacts here:

https://github.com/python/release-tools/blob/401afdec47eadf4b5f17a68f5bb87f4ca04a86d1/run_release.py#L574-L578

So we need changes to the `add_to_python_dot_org.py` script to correctly change the permissions as well.

**Comment by pablogsal** [comment]
Given that this is impacting releases, I am reverting the sigstore changes in the repository until they are ready.

**Comment by pablogsal** [comment]
I also fixed manually the permissions for 3.10.7

**Comment by di** [comment]
Thanks, I filed https://github.com/python/release-tools/issues/21 to capture what (I think) needs done to fix this.

**Pull request merged**
